design/cluster-scope-resource-filter.md

```diff
@@ -0,0 +1,572 @@
+
+	// IncludedClusterScopeResources is a slice of cluster scope 
+	// resource type names to include in the restore.
+	// If not specified, all cluster scope resources in the backup are included.
+	// Supersede IncludedResources in conflict case.
+	// +optional
+	// +nullable
+	IncludedClusterScopeResources []string `json:"includedClusterScopeResources,omitempty"`
+
+	// ExcludedClusterScopeResources is a slice of cluster scope 
+	// resource type names to exclude from the restore.
+	// Supersede ExcludedResources in conflict case.
+	// +optional
+	// +nullable
+	ExcludedClusterScopeResources []string `json:"excludedClusterScopeResources,omitempty"`
+
+	// IncludedNamespacedResources is a slice of namespace scope
+	// resource types names to include in the restore.
+	// If not specified, all namespaced scope resources in the backup are included.
+	// Supersede IncludedResources in conflict case.
+	// +optional
+	// +nullable
+	IncludedNamespacedResources []string `json:"includedNamespacedResources,omitempty"`
+
+	// ExcludedNamespacedResources is a slice of namespace scope
+	// resource type names to exclude from the backup.
+	// Supersede ExcludedResources in conflict case.
+	// +optional
+	// +nullable
+	ExcludedNamespacedResources []string `json:"excludedNamespacedResources,omitempty"`
+    ......
+}
+```
+
+## Alternatives Considered
+Proposal from Jibu Data [Issue 5120](https://github.com/vmware-tanzu/velero/issues/5120#issue-1304534563)
+
+
+## Security Considerations
+No security impact.
+
+## Compatibility
+### IncludedResources and ExcludedResources change
+To make the change backward compatible, `IncludedResources/ExcludedResources` parameters still apply to both cluster scope and namespace scope.
+After adding new parameters `IncludedClusterScopeResources`, `ExcludedClusterScopeResources`, `IncludedNamespacedResources` and `ExcludedNamespacedResources`, there could be conflict between the exiting and new parameters.
+When there is conflict, `IncludedClusterScopeResources/ExcludedClusterScopeResources` parameters have higher priority for cluster scope resources.
+`IncludedNamespacedResources/ExcludedNamespacedResources` parameters have higher priority for namespace scope resources.
+
+### IncludeClusterResources change
+`IncludeClusterResources` parameter is an existing boolean parameter, which implies whether to include cluster scope resources.
+There are three possible values of this parameter.
+In previous release, not setting value means including cluster scope resources related to namespaced scope resources included in backup and restore; `true` means including all cluster resources; `false` means excluding all cluster resources.
+
+After this change, `IncludeClusterResources`'s not setting value and `false`'s meaning are not changed.
+If `IncludeClusterResources` is set to `true`, and `IncludedClusterScopeResources/ExcludedClusterScopeResources` are not set, all cluster resources are included.
```

Comment:
I think this particular one introduces confusion. It will be simpler if both IncludedClusterScopeResources and IncludedNamespacedResources must always be present with a value of "*" to include all, otherwise we assume none are included.